sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -347,6 +347,14 @@ querySpecification
        windows?)
     ;
 
+hint
+    : '/*+' mapJoinHint '*/'
+    ;
+
+mapJoinHint
+    : MAPJOIN '(' broadcastedTables+=tableIdentifier (',' broadcastedTables+=tableIdentifier)* ')'
```

Comment:
Can we move the logic somewhere else that's not in the parser rule? It'd be great if we don't need to modify the parser rule in order to support "broadcast" or "broadcast join". Even moving it into the parser itself (that is not parser rule could work).